x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/mapper/HistogramFieldMapper.java

```
@@ -0,0 +1,459 @@
+                                        throw new IllegalStateException("Cannot load doc values", e);
+                                    }
+
+                                }
+
+                                @Override
+                                public ScriptDocValues<?> getScriptValues() {
+                                    return new ScriptDocValues.Strings(getBytesValues());
+                                }
+
+                                @Override
+                                public SortedBinaryDocValues getBytesValues() {
+                                    try {
+                                        final BinaryDocValues values = DocValues.getBinary(context.reader(), fieldName);
+                                        return FieldData.singleton(values);
+                                    } catch (IOException e) {
+                                        throw new IllegalStateException("Cannot load doc values", e);
+                                    }
+                                }
+
+                                @Override
+                                public long ramBytesUsed() {
+                                    return 0; // Unknown
+                                }
+
+                                @Override
+                                public void close() {
+
+                                }
+                            };
+                        }
+
+                        @Override
+                        public AtomicHistogramFieldData loadDirect(LeafReaderContext context) throws Exception {
+                            return load(context);
+                        }
+
+                        @Override
+                        public SortField sortField(Object missingValue, MultiValueMode sortMode,
+                                                   XFieldComparatorSource.Nested nested, boolean reverse) {
+                            return null;
```

this should throw an UnsupportedOperationException too